shared/js/content-scripts/click-to-load.js

```diff
@@ -812,6 +1105,51 @@
                 titleRowTextButton.style.display = 'block'
             }
         }
+
+        /** YouTube CTL */
+        if (widget.replaceSettings.type === 'youtube-video') {
+            await toggleYouTubeCTL(trackingElement, widget)
+
+            /** TODO
+             *  Listen to `ddg-settings-youtubePreviewsEnabled` message to update YT CTL state
+             * (ie switch between blocked and preview modes)
+             */
+        }
+    }
+
+    /**
+     * @param {Element} trackingElement
+     *   The original tracking element (YouTube video iframe)
+     * @param {DuckWidget} widget
+     *   The CTP 'widget' associated with the tracking element.
+     */
+    async function toggleYouTubeCTL (trackingElement, widget) {
+        const youtubePreviewsEnabled = await getYouTubePreviewsEnabled()
+
+        // Show YouTube Preview for embedded video
+        if (youtubePreviewsEnabled === true) {
+            const { placeholder } = await createYouTubePlaceholder(trackingElement, widget)
+            replaceTrackingElement(
+                widget, trackingElement, placeholder, /* hideTrackingElement= */ true
+            )
+
+        // Block YouTube embedded video and display blocking dialog
+        } else {
+            const { blockingDialog, shadowRoot } = await createYouTubeBlockingDialog(trackingElement, widget)
+            replaceTrackingElement(
+                widget, trackingElement, blockingDialog, /* hideTrackingElement= */ true
+            )
```

Comment:
@kzar This should fix the issue of YT details being requested. I was creating the preview placeholder and the blocking dialog elements before checking what setting was enabled, causing to request video details even when not necessary.